### PR TITLE
Adding StudySession to Tutor-Tutee-Match Algorithm

### PR DESCRIPTION
**Pull request by DhruvRelwani**
Pivotal Tracker IDs : [152774436, 152766340]

Changes: 
tl;dr : Assigned a specific time as a part of the tutor-tutee match.
1) Added a StudySession model which contains 1 tutor, 1 tutee and 1 time_availability. 
2) Added 3 migrations which create the study_sessions table, add ids to existing tables and create the study_sessions_time_availabilities table.
3) Changed the tutor_match page to show the assigned study session time and the tutor.
4) Changed a few of the dependent cucumber tests to align with these new changes.

### CODE REVIEW

File: app/models/study_session.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class StudySession < ActiveRecord::Base
+	has_and_belongs_to_many :time_availabilitys, as: :time_availabilityable
```
Comment (junomorey) [inline app/models/study_session.rb:2]: Can this just be has_one?

Comment (DhruvRelwani) [reply app/models/study_session.rb:2]: It had the same issue as adding a time availability to a tutor and a tutee (when we add ta_object1 for 1 session, it was taken off from another session). We have `has_and_belongs_to_many` to allow multiple pairs of tutors and tutees to meet at a single time. i.e. tutee1 and tutor1 can meet at 8am on Monday and so can tutee2 and tutor2. In practice however, we never add more than 1 time_availability to a session for a particular pair of tutor and tutee.

Comment (apolevoi) [reply app/models/study_session.rb:2]: So to confirm, even though there is only one time availability per session, this still has to be plural?

**Review by apolevoi — COMMENTED** [review]
How come in find_tutor.feature you had to change which tutees show up in a search? Is this because now we are using only the first ta? Is @study_session_time the decided time?


Reply:
The tests had been written in a way that previously allowed more than 1 tutee to attend a session with a particular tutor at a specific time. I changed the seed file and the tests to resolve that. Now, tutee3 would be assigned no tutors (because of its time availabilities) and tutee4 will be assigned to tutor1. To get a better understanding, simply check the comment on top of these user stories (there's a table about the time availabilities for each tutor and tutee).

File: app/models/study_session.rb (hunk 1 above)
Comment (DhruvRelwani) [reply app/models/study_session.rb:2]: Yes, that's correct. It's to resolve the problem described in the previous comment

Comment (apolevoi) [reply app/models/study_session.rb:2]: I think this is ready to be pushed if Juno agrees. Do you want to wait for him to sign off too?

**Review by junomorey — APPROVED** [review]

**Pull request merged**